### ISSUE TITLE
Connected pull requests information

### ISSUE THREAD

**Issue by fiddymaster** [rfennell/AzurePipelines#624]
### Azure DevOps Extension you are using
Using the old extension, but do not mind switching to new one.

### Where are you running it?
- Azure DevOps Service (VSTS)

### Version of Extension/Task
3.6.1

### Question
Because i cant find another way to ask you a question, i am creating this issue.

I want to know how to add pull request information to the release notes. I have looked in the code and on msdn to see what api calls there are, but cant find an easy way to do this. I hope you know how to help me with this.

The thing i want to achieve is:
@@WILOOP@@  
<li>
$($widetail.fields.'System.WorkItemType') $($widetail.id): <a href="https://dev.azure.com/teamname/projectName/_workitems?id=$($widetail.id)" target="_blank">$($widetail.fields.'System.Title')</a>
<ul><li><a href="https://dev.azure.com/divideteam/Divide.NOW/_git/DivideCore/pullrequest/$($widetail.fields.'System.PullRequest.PullRequestId')" target="_blank">Connected PR</a></li></ul>
</li>
@@WILOOP@@

I hope this is possible, if not our customer support just needs to click a bit more :).

**Comment by rfennell** [comment]
I can only think of two ways

1. loads of extra handling - doable but will need work
2. make the assumption that the branch name matches the PR as it will if the pipeline is being used for branch validation and juts use the primary source artifact branch

**Comment by rfennell** [comment]
I have been having a look at this and made a change to get close to what you asked without a very major re-write.

The problem is that calls used to get the commits and work items are getChangesBetweenBuilds(B1, B2) and getWorkitemsBetweenBuilds(B1, B2). These return object arrays with the changes and work items as you would expect.

The problem is that the PR details are only available in the build.triggerInfo, not the WI or changes.

The change I have made is that when running in a release, I now expose the build that caused the release to trigger as an object. So in a template you can call

```
**Build Trigger PR Number**: ${buildDetails.triggerInfo['pr.number']}
```
If you use this you get back a number if the trigger was a PR or a 'undefined' if not. Unfortunately you don't get a URL, but you could build this up from other properties.

The advantage of this model is that I am exposing a standard object so you get any data associated. To dump the log set

```
${JSON.stringify(buildDetails)}    
```
This means if you have some other trigger you could use that in the release notes too

This has been released in 2.26.1

**Comment by rfennell** [comment]
Thanks for the PR, interesting idea.


I can see that is going to take a while to have a look at. I can see for a start that the [CI build](https://richardfennell.visualstudio.com/GitHub/_build/results?buildId=4432&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=295ef4c3-3a46-5f11-020b-b8098f9465c0) has failed, so I need to check that.

Do you have a sample handbars template  can add to my release pipeline as a integration test?
